library/catalog/templates/login_redirect.html

```diff
@@ -0,0 +1,122 @@
+    <meta http-equiv="x-ua-compatible" content="ie=edge"/>
+    <link rel="stylesheet" href="{% static 'css/main.css' %}">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css">
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+    <meta charset="UTF-8">
+    <title>Login Error</title>
+</head>
+<body>
+{% csrf_token %}
+
+<button data-toggle="modal" data-target="#authModal" data-toggle="modal" type="submit" class="btn btn-primary">Sign in
+</button>
+
+
+<div id="authModal" class="modal fade loginM" role="dialog">
+    <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-dialog" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="AuthModalLabel">Sign In</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">×</span>
+                    </button>
+                </div>
+
+
+                <div class="modal-body">
+                    <form class="login-form">
+
+                    <div class="form-group">
+                        <label for="inputUsername">Username</label>
+                        <input type="email" class="form-control" id="inputUsername"
+                               aria-describedby="emailHelp" placeholder="Username">
+                        <small id="emailHelp" class="form-text text-muted">Enter your username</small>
+                    </div>
+
+                    <div class="form-group">
+                        <label for="inputPassword">Password</label>
+                        <input type="password" name="password" class="form-control" id="inputPassword"
+                               placeholder="Password">
+                        <div id="errorlogin" style="color:red"></div>
+                    </div>
+
+                    </form>
+                </div>
+
+                <div class="modal-footer">
+
+                    <div class="mb-3">
+                        <a href="{% url 'library-registration' %}" class="btn btn-link">Sign Up</a>
+                    </div>
+
+                    {% if request.GET.next %}
+                    <input type="hidden" name="next" id="nextPage" value="{{request.GET.next}}">
+                    {% endif %}
+
+                    <div class="mb-3">
+                        <button type="button" class="btn btn-primary btn-lg btn-block login-btn"
+                                style="max-width:500px" onclick="login_user()">Sign In
+                        </button>
+                    </div>
+
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+{% if request.GET.next %}
+<input type="hidden" name="next" value="{{request.GET.next}}">
+{% endif %}
+
+
+<!-- Custom scripts -->
+<script type="text/javascript">
```

Comment:
Can we put this code in the sepereate file ?

Reply:
> Can we put this code in the sepereate file ?

I don't know because I use jinja for building a full path to my login function ( url:'{% url 'func_login' %}',) where JS send a request with credentials
If I put this code in a separate file, I will have to write the full path this way (library / func_login). I think this can create problems if we change the program architecture. 
And now I have trouble with the redirection page for this reason.
I am researching this question and will be able to give a better answer in the near future.